### ISSUE TITLE
[Publications] MozFest specializations 

### ISSUE THREAD

**Issue by kristinashu** [mozilla/foundation.mozilla.org#5160]
[placeholder ticket]

**Comment by kristinashu** [comment]
We met with Marc and Hey Hey. Kalob shared where we're at with the publications template build and Hey Hey shared their vision. Next steps, we will meet with directors to learn what the goals of the book are and how important the book is in the context of the entire festival work. In the meantime, Kalob will look into some existing wagtail examples to see if they could work for Hey Hey's vision of the book.

**Comment by KalobTaulien** [comment]
### Small animations 
For small little animations we can leverage animate[animate.css](https://animate.style/) - these are all simple to implement as it's all CSS. We'd just need to use a little JavaScript  (`.scrollIntroView()` [more here](https://developer.mozilla.org/en/docs/Web/API/Element/scrollIntoView)) for this to operate smoothly. 

### Shareable text on a page. 
For highlight text on the page and making it shareable, I found this cool (and small) library called [Sharect](https://estevanmaito.github.io/sharect/). I _feel_ like there's more behind the scenes than what I've thought about here, such as making the text highlighted on the page when someone clicked the shared link, and other related situations. 

### Interview conversations 
Adding playful interview text could be easy using animate.css and `.scrollIntoView()` I think. By "easy" I mean simple. That could be a streamfield with a listblock of structblocks along with an option to be "on the left" or "on the right". As they are scrollde into view they can pop up. Though I'll be honest, I don't think that's great for accessibility to have features hidden and then show up when they appear on the page. Screen readers will likely have a fit over that. 

### Dark mode 
Light and dark mode would, in theory, not be hard to implement either. If we've been using bootstrap properly (we put in a solid effort to do this) then dark mode should be relatively straightforward to implement. I'm not convinced this adds additional value to the reader, though. We've seen "dark mode" implemented in almost every app these days, and before it was a common thing almost nobody ever said "I wish _this website_ had a dark mode". We'd need to implement opposite-colored images too, and that may become a heartbreaker for the team. We could chat more about this though. 

### Individual chapter design
Chapters having their own visual design would be hard to implement without giving the editors access to raw html/css/js fields. While exposing those fields would be simple, it's a frowned upon feature in case someone accidentally XSS attacks themselves, and goes against the wagtail principle of "keeping editing simple". A better solution would be to toggle on/off certain features with checkboxes or select an option from a dropdown. That limits the chapter customizability but also allows each chapter to have its own custom styling. 

### Bookmarking/Dog Earing
"Dog earing" pages seems like it's identical to the browsers' native bookmark feature. I don't see reader value in that. 

### Page progress 
Page progress (the bar the moves from left to right as you progress through an Article) would be relatively simple to implement. [Here is a good example](https://codepen.io/jbsmith731/pen/NPmYEJ) of it, though this is using jQuery migrating this to vanilla JS wouldn't be a crazy task either. From previous experience, I know a lot of users who read a long article like to see their progress in this visual way. There's also a tonne of small libraries we could implement instead of hand-rolling our own, too. 

### Continue where you left off 
"Continue where you left off" is a cool idea, I really like that one for long read that span several (or hundreds) of pages. We could use JavaScript's `localStorage` for this, however we'll want to run this by the legal team I think. Implementation would be to remember the latest page the user was on, likely by Page.depth. We'd want a way to turn this off for people who, say, read Article page 90 at first, but decided to start at the beginning of the chapter. That would require more `localStorage`, and again should be run by the legal team imo. 

### Timeline(s)
For timelines, I've used [timeline.js](https://timeline.knightlab.com/) before and it was easy enough to work with. Though we probably want the timeline to skip the content section rather than scroll the timeline and I dont know what that would take with this library. An alternative idea is a menu along the side of the page that can skip to certain streamfield sections. I think this feature is going to require more conversation about how we want it to act before we can determine how we'd create it (a UX/UI conversation). 

### Marking/Highlighting multiple areas and "saving" them 
Marking (selecting and highlighting multiple sentences in different areas) could be interesting and _possibly_ a decent value-add to the reader. But going as far as saving a link for them could end up in A LOT of old saved data bloating the database — this should be a conversation with Cade and UX/Deisgn people about long term implementation. We'd also need to leverage `localStorage` which means involving the legal team. We'd likely use a highlighting javascript library, and after each highlight "save" the code in a massive text field (maybe a JsonField). Then we'd need a way to pull the saved code from the database, check for a match in the content and re-highlight the content on the page. I feel like this would be a tremendous amount of work for little payoff this year, but again we should consider a larger conversation around this one.

**Comment by Pomax** [comment]
Moving this to stretch for now, feel free to move back when we finish up the PNI work early

**Comment by kristinashu** [comment]
Closing this. Still not sure what the mozfest team has decided but will open a new ticket when they do.
